data/crac/crac-api/src/main/java/com/powsybl/openrao/data/cracapi/Crac.java

```diff
@@ -531,4 +537,71 @@ default State getState(String contingencyId, Instant instant) {
      * Get a {@link RaUsageLimitsAdder}, to add a {@link RaUsageLimits} to the crac
      */
     RaUsageLimitsAdder newRaUsageLimits(String instantName);
+
+    /**
+     * Import CRAC from a file
+     *
+     * @param importers   candidates CRAC importers to process the data
+     * @param inputStream CRAC data
+     * @param cracFactory CRAC factory
+     * @param network     the network on which the CRAC data is based
+     * @return CRAC object
+     */
+    static Crac read(List<Importer> importers, InputStream inputStream, CracFactory cracFactory, Network network) {
+        for (Importer importer : importers) {
+            try {
+                return importer.importData(inputStream, cracFactory, network);
+            } catch (Exception ignored) {
+            }
```

Comment:
instead of try/catch, use "exists" (that you shouldn't have removed)

Reply:
previously the `exists` methods checked that the file was a JSON file but this new method does not require a filename do you have an idea how to do this?

Reply:
I pushed a new version using `exists` that works but imo it is a rather bad way to do, what do you think?

Reply:
> I pushed a new version using `exists` that works but imo it is a rather bad way to do, what do you think?

I will wait for your feedback before I do the same for RaoResut

Reply:
can't you use the same approach as NAtiveCracImporters.ImportData ? this would avoid the need of storing a "data" attribute.